client/asset/btc/btc.go

```diff
@@ -1362,6 +1407,43 @@ func (btc *baseWallet) feeRate(_ RawRequester, confTarget uint64) (uint64, error
 	return uint64(dex.IntDivUp(int64(satPerKB), 1000)), nil
 }
 
+// externalFeeEstimator gets the fee rate from the external API
+func externalFeeEstimator(ctx context.Context, net dex.Network, nb uint64) (float64, error) {
+	var url string
+	if net == dex.Testnet {
+		url = testnetExternalApiUrl
+	} else {
+		url = externalApiUrl
+	}
+	url = url + "v1/fees/recommended"
+	ctx, cancel := context.WithTimeout(ctx, 4*time.Second)
+	defer cancel()
+	r, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+
+	if err != nil {
+		return 0, err
+	}
+	httpResponse, err := client.Do(r)
+	if err != nil {
+		return 0, err
+	}
+	var resp map[string]uint
+	reader := io.LimitReader(httpResponse.Body, 1<<20)
+	err = json.NewDecoder(reader).Decode(&resp)
+	if err != nil {
+		return 0, err
+	}
+	httpResponse.Body.Close()
+
+	// we use fastestFee, docs to mempool api https://mempool.space/docs/api
+	feeInSat, ok := resp["fastestFee"]
+	if !ok {
+		return 0, errors.New("no fee rate found")
+	}
+	estimatedFeeRate := btcutil.Amount(feeInSat).ToBTC()
```

Comment:
`feeInSat` is sat **per vByte**, so doing `btcutil.Amount(feeInSat).ToBTC()` gives BTC/B rather than BTC/kB